src/openrct2-ui/windows/ScenarioSelect.cpp

```diff
@@ -766,29 +765,30 @@ static void InitialiseListItems(WindowBase* w)
                     rct1CompletedScenarios |= 1 << scenario->ScenarioId;
                 }
             }
-
-            // If scenario is Mega Park, keep a reference to it
-            if (scenario->ScenarioId == SC_MEGA_PARK)
-            {
-                megaParkListItemIndex = _listItems.size() - 1;
```

Comment:
Mega Park was not yet added to the list here, so the index was actually off-by-one.